### ISSUE TITLE
Add server difficulty enviornment variables?

### ISSUE THREAD

**Issue by jonneymendoza** [mornedhels/enshrouded-server#72]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

A new update version 0.7.3.0 has released which contains a new feature where you can tweak the server difficulty.

Is it possible to add the envi variables so that we can tweak the server difficulty as mentioned here?

https://enshrouded.zendesk.com/hc/en-us/articles/20453241249821-Server-Difficulty-Settings

Thanks

**Comment by cp-fabian-pittroff** [comment]
Hey @jonneymendoza,

thanks for the suggestion. I'll have a look at the weekend. For the time beeing, you can edit the `enshrouded_server.json` manually and configure the server difficulty that way. Just make sure you stop the current running container before editing the file. It should not overwrite it, but to be sure.

path: `game/server/enshrouded_server.json`

The server should automatically add those settings on start. Also if you want to use your custom settings, you have to switch the preset to `Custom`.

shortened example:
```json
{
  ...
  "slotCount": 16,
  "gameSettingsPreset": "Default",
  "gameSettings": {
    "playerHealthFactor": 1,
    "playerManaFactor": 1,
    "playerStaminaFactor": 1,
    ...
  }
}
```

**Comment by cp-fabian-pittroff** [comment]
Do you have a suggestion on how to name the environment envs?

I'm leaning to:
* `GS_PRESET`
* `GS_PLAYER_HEALTH_FACTOR`
* `GS_PLAYER_MANA_FACTOR`
* ...

**Comment by jonneymendoza** [comment]
Yea that makes sense to do it like that.



> Do you have a suggestion on how to name the environment envs?
> 
> I'm leaning to:
> 
> * `GS_PRESET`
> * `GS_PLAYER_HEALTH_FACTOR`
> * `GS_PLAYER_MANA_FACTOR`
> * ...
>   GS for game settings

**Comment by jonneymendoza** [comment]
> Hey @jonneymendoza,
> 
> thanks for the suggestion. I'll have a look at the weekend. For the time beeing, you can edit the `enshrouded_server.json` manually and configure the server difficulty that way. Just make sure you stop the current running container before editing the file. It should not overwrite it, but to be sure.
> 
> path: `game/server/enshrouded_server.json`
> 
> The server should automatically add those settings on start. Also if you want to use your custom settings, you have to switch the preset to `Custom`.
> 
> shortened example:
> 
> ```json
> {
>   ...
>   "slotCount": 16,
>   "gameSettingsPreset": "Default",
>   "gameSettings": {
>     "playerHealthFactor": 1,
>     "playerManaFactor": 1,
>     "playerStaminaFactor": 1,
>     ...
>   }
> }
> ```

Ahh ok cool i will try that. i did not know you can persist the config as i thought that gets auto created based on my compose yaml file

**Comment by cp-fabian-pittroff** [comment]
the container automatically updates the server via steam. To stop the container, just use `docker compose down` edit the file inside `/home/config/enshrouded/server/enshrouded_server.json` and start it back up with `docker compose up -d`

**Comment by jonneymendoza** [comment]
Thanks. I will try that out later

**Comment by cp-fabian-pittroff** [comment]
released with [1.3.0](https://github.com/mornedhels/enshrouded-server/releases/tag/1.3.0) - env vars are prefixed with `SERVER_GS_`

**Comment by jonneymendoza** [comment]
Awsome
